### ISSUE TITLE
Control conduit tick rates

### ISSUE THREAD

**Issue by Shybella** [SleepyTrousers/EnderIO#4790]
#### Issue Description: I have a large server and I'm experiencing large 15-16% impact from EnderIO conduits. Spark: http://prntscr.com/k9990g
____
#### Affected Versions (Do *not* use "latest"):

- EnderIO: EnderIO-1.12.2-5.0.29
- EnderCore: EnderCore-1.12.2-0.5.35
- Minecraft: 1.12.2
- Forge: 2705

I can provide for details/logs/profiles and even snapshots if needed. Adding a configurable tick rate to conduits would be ideal in my case while I am able to increase the output per tick lowering the tick rate and increasing the output should reduce performance impact. 

Possibly there is a optimization problem and something can be improved. 

Thanks.

**Comment by HenryLoenwind** [comment]
At the moment it only runs every 20 ticks, I doubt having it run less often would change anything but produce stuttering...

I recommend you tell your players to not build so many automations that hammer needlessly against machines that won't ever accept the input.

**Comment by Shybella** [comment]
Yeah, if it's already limited to running once every 20 ticks this will not help. 

EnderIO performs the best that I've seen when it comes to conduits. This maybe just a limitation to what Minecraft can handle. 

Thanks.

**Comment by Shybella** [comment]
Is there any type of optimization that can be done? I've reduced conduits down from 5000 to 3000 on the server but ticking harder now at 33% avg on tick loop time. 

I can provide profiles, snapshots anything to help if possible.

**Comment by HenryLoenwind** [comment]
In your screenshot, a good half of the tick time was spent inside the code for that one generator. I'd recommend to have a look if it's just that one generator or if there are others eating up time, too. If it's just that generator, then it's most likely a single setup using all that cpu. If it's multiple machines, I'd recommend to ask their authors in a friendly(!!) way if they see any chance of optimizations.

Also, I put something into our conduits (next release...) that may help in this specific setup, making them sleep for a longer (and configurable) time if they fail to transfer any item.

**Comment by Shybella** [comment]
Thanks for the reply.

Thanks for looking into the issue and making an attempt to improve my situation. If you need testing done on a map with 3000+ conduits, I'm able to help.

**Comment by HenryLoenwind** [comment]
sure, if you want to run a test with the current dev build ( http://ci.tterrag.com/job/EnderIO-Modules/job/EnderIO-Hourly/595/ ) that would be nice. Just don't throw it onto your real server; switching back to the release version isn't something that necessarily works.

**Comment by EpicSquid** [comment]
If you want to follow up with testing, try using our discord. Closing this issue
